StickyFooter

ckgraybeal opened the pull request:
@dbarbalato

Hey David,

I've added the sticky footer functionality as a bool prop to be passed into Sticky. Please let me know if you have any comments. Thanks again for considering this!

Chris

dbarbalato commented on src/Sticky.js:
```diff
@@ -8,15 +8,17 @@ export default class Sticky extends Component {
     relative: PropTypes.bool,
-    children: PropTypes.func.isRequired
+    children: PropTypes.func.isRequired,
+    isFooter: PropType.bool
```
First, it's PropTypes, not PropType.

Second, I think `isFooter` is a bit implementation specific -- I'd rather see a prop called `edge` that accepts a couple of exported constants, (e.g. `export Edges = { TOP: 0, BOTTOM: 1 }`, and defaults to `Edges.TOP`

ckgraybeal replied:
yep, makes sense for both.  Will make this update

dbarbalato commented on src/Sticky.js:
```diff
@@ -40,41 +42,60 @@ export default class Sticky extends Component {
 
-    const wasSticky = !!this.state.isSticky;
-    const isSticky = preventingStickyStateChanges ? wasSticky : (distanceFromTop < -this.props.topOffset && distanceFromBottom > -this.props.bottomOffset);
+  const wasSticky = !!this.state.isSticky;
```
Fix spaces

dbarbalato commented on src/Sticky.js:
```diff
@@ -40,41 +42,60 @@ export default class Sticky extends Component {
-      width: placeholderClientRect.width,
-      transform: this.props.disableHardwareAcceleration ? '' : 'translateZ(0)'
-    }
+  if (!this.props.isFooter) {
```
How does this work when `this.props.relative` is `true`?

ckgraybeal replied:
hmmm definitely doesnt work for relative yet.  Will make changes for this as well

dbarbalato commented:
It's a cool feature - nice job. 

I would like to see a dedicated 'bottom' example, and an addition to the README detailing how this works before considering a merge.

Additionally, this will need tests once the existing tests are re-written. If you wanted to take a stab at the rewrite of tests to get your feature in, by all means go ahead. Otherwise, I will keep this open until the new tests are in place, at which point this PR will get merged (pending the above conditions as well).

Thanks!

ckgraybeal commented:
Hey Dave,

I've addressed most of your comments.  However, I am having a really difficult time applying my changes to a `relative` sticky.  Do you mind taking a look at the current `Edge.BOTTOM` path and pointing me in the right direction?

As for tests,  I had tests written but removed them since the whole suite was broken.  I don't have the time to tackle the test suite right now, but will be happy to add them later.

robbiemccorkell commented:
Does this achieve the same as https://github.com/captivationsoftware/react-sticky/pull/132

It looks like that PR was merged into a branch and the feature still needs some tests updated before it can go in.

dbarbalato commented:
To my understanding, yes it does.

It is in progress but, as stated above, it still needs some finishing work. I don't have the time to do it, so if someone else wants to take initiative w/ tests and the like, we can move ahead with it.
